test/access/manager/AccessManager.test.js

```diff
@@ -1670,6 +1671,60 @@ describe('AccessManager', function () {
     });
   });
 
+  describe('access managed self operations', function () {
+    describe('when calling a restricted target function', function () {
+      beforeEach('set required role', function () {
+        this.method = this.target.fnRestricted.getFragment();
+        this.role = { id: 785913n };
+        this.manager.$_setTargetFunctionRole(this.target, this.method.selector, this.role.id);
```

Comment:
Should we set the function role for `this.manager, this.method.selector` ? It feels like we are testing restricted operation on the target when we should test the restricted operation on the manager.